immuni_common/helpers/utils.py

```diff
@@ -52,3 +53,32 @@ def dense_dict(dictionary: Dict) -> Dict:
         for k, v in dictionary.items()
         if v is not None
     }
+
+
+# Note: This could be a Generic[T] with T as TypeVar. That would allow
+#  us to define a WeightedPair[str] and have the weighted_random
+#  return a valid typed str when used, but generic NamedTuples are not yet
+#  supported by mypy:
+#  https://github.com/python/mypy/issues/685
+class WeightedPair(NamedTuple):
+    """
+    Simple NamedTuple that allows us to describe
+    weighted key/value pairs for weighted random.
+    """
+
+    weight: int
+    payload: Any
+
+
+def weighted_random(pairs: List[WeightedPair]) -> Any:
```

Comment:
I'd change `pairs` to `responses` and `WeightedPair` to `WeightedResponse`.
The only reason I'd stick to pairs is "we have plans to use it for anything, not strictly responses" (which I don't think is the case right now.) 🤔